code/game/objects/structures/barsign.dm

```diff
@@ -47,68 +52,141 @@
-			to_chat(user, "<span class='notice'>You close the maintenance panel.</span>")
-			if(!broken && !emagged)
-				set_sign(pick(barsigns))
-			else if(emagged)
-				set_sign(new /datum/barsign/hiddensigns/syndibarsign)
-			else
-				set_sign(new /datum/barsign/hiddensigns/empbarsign)
-			panel_open = 0
-
+/obj/structure/sign/barsign/attackby(obj/item/I, mob/user)
 	if(istype(I, /obj/item/stack/cable_coil) && panel_open)
 		var/obj/item/stack/cable_coil/C = I
-		if(emagged) //Emagged, not broken by EMP
+		if(broken && emagged) //Emagged, not broken by EMP
 			to_chat(user, "<span class='warning'>Sign has been damaged beyond repair!</span>")
 			return
-		else if(!broken)
+		else if(broken && !emagged)
+			if(C.use(2))
+				to_chat(user, "<span class='notice'>You replace the burnt wiring.</span>")
+				broken = FALSE
+			else
+				to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+		else if(!broken && !emagged)
 			to_chat(user, "<span class='warning'>This sign is functioning properly!</span>")
 			return
-
-		if(C.use(2))
-			to_chat(user, "<span class='notice'>You replace the burnt wiring.</span>")
-			broken = 0
-		else
-			to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+	if(isscrewdriver(I) || iswrench(I))
+		return	// No damage.
 	else
 		return ..()
 
+/obj/structure/sign/barsign/screwdriver_act(mob/user)
+	if(!panel_open)
+		to_chat(user, "<span class='notice'>You open the maintenance panel.</span>")
+		set_sign(new /datum/barsign/signoff)
+		panel_open = TRUE
+	else
+		if(prev_sign.icon != icon_state && istype(prev_sign, /datum/barsign))	// Sanity, I suppose
+			set_sign(prev_sign)
+		to_chat(user, "<span class='notice'>You close the maintenance panel.</span>")
+		panel_open = FALSE
+	return
+
+/obj/structure/sign/barsign/wrench_act(mob/user)
+	if(!panel_open)
+		to_chat(user, "<span_class='notice>You must first open the maintenance panel before trying to unbolt the barsign.</span>")
+		return
+	else
+		var/obj/item/sign/barsign/S = new(user.loc)
+		S.name = name
+		S.desc = desc
+		S.icon_state = icon_state	//The only sprite direction that exists is South.
+		S.broken = broken
+		S.emagged = emagged
+		S.req_access = req_access
+		S.panel_open = !panel_open
+		S.prev_sign = prev_sign
+		qdel(src)
+
+/obj/item/sign/barsign
+	name = "barsign"
+	desc = ""
+	icon = 'icons/obj/barsigns.dmi'
+	w_class = WEIGHT_CLASS_HUGE		// Don't put this in your backpack!
+	resistance_flags = FLAMMABLE
+	var/panel_open
+	var/broken
+	var/datum/barsign/prev_sign
+
+/obj/item/sign/barsign/wrench_act(mob/user)	//construction
+	if(isturf(user.loc))
+		var/direction = input("In which direction?", "Select direction.") in list("North", "East", "South", "West", "Cancel")
+		//The only sprite that exists is South, but we forced this in when we dismantled the barsign.
+		if(direction == "Cancel")
+			return
+		if(QDELETED(src))
+			return
+		var/obj/structure/sign/barsign/S = new(user.loc)
```

Comment:
Be careful with this. The user can still move/be moved during the input and since it's technically blocking, their `loc` could've changed in the meantime and could be a locker/space pod. Please add a turf check.